cli/src/run_testenv.rs

```diff
@@ -0,0 +1,396 @@
+        create_new_device(new_device.clone(), author, now);
+
+    match cmds
+        .send(device_create::Req {
+            device_certificate,
+            redacted_device_certificate,
+        })
+        .await
+    {
+        Ok(rep) => match rep {
+            DeviceCreateRep::Ok => (),
+            _ => return Err(anyhow::anyhow!("Cannot create device: {rep:?}")),
+        },
+        Err(e) => return Err(anyhow::anyhow!("{e}")),
+    }
+
+    let key_file = libparsec::get_default_key_file(&client_config.config_dir, &new_device);
+
+    let access = DeviceAccessStrategy::Password {
+        key_file,
+        password: DEFAULT_DEVICE_PASSWORD.to_string().into(),
+    };
+
+    libparsec::save_device(Path::new(""), &access, &new_device).await?;
+
+    Ok(new_device)
+}
+
+async fn register_new_user(
+    client_config: &ClientConfig,
+    cmds: &AuthenticatedCmds,
+    author: Arc<LocalDevice>,
+    device_label: DeviceLabel,
+    human_handle: HumanHandle,
+    initial_profile: UserProfile,
+) -> anyhow::Result<Arc<LocalDevice>> {
+    let new_device = Arc::new(LocalDevice::generate_new_device(
+        cmds.addr().clone(),
+        initial_profile,
+        human_handle,
+        device_label,
+        None,
+        None,
+        None,
+    ));
+    let now = author.now();
+
+    let (user_certificate, redacted_user_certificate) =
+        create_new_user(new_device.clone(), author.clone(), initial_profile, now);
+    let (device_certificate, redacted_device_certificate) =
+        create_new_device(new_device.clone(), author, now);
+
+    match cmds
+        .send(user_create::Req {
+            device_certificate,
+            redacted_device_certificate,
+            redacted_user_certificate,
+            user_certificate,
+        })
+        .await
+    {
+        Ok(rep) => match rep {
+            UserCreateRep::Ok => (),
+            _ => return Err(anyhow::anyhow!("Cannot create user: {rep:?}")),
+        },
+        Err(e) => return Err(anyhow::anyhow!("{e}")),
+    }
+
+    let key_file = libparsec::get_default_key_file(&client_config.config_dir, &new_device);
+
+    let access = DeviceAccessStrategy::Password {
+        key_file,
+        password: DEFAULT_DEVICE_PASSWORD.to_string().into(),
+    };
+
+    libparsec::save_device(Path::new(""), &access, &new_device).await?;
+
+    Ok(new_device)
+}
+
+async fn initialize_test_organization(
+    client_config: ClientConfig,
+    addr: BackendAddr,
+) -> anyhow::Result<[Arc<LocalDevice>; 3]> {
+    let organization_id: OrganizationID = "Org".parse().expect("Unreachable");
+
+    // Create organization
+    let bootstrap_token = loop {
+        match create_organization_req(&organization_id, &addr, DEFAULT_ADMINISTRATION_TOKEN).await {
+            Ok(bootstrap_token) => break bootstrap_token,
+            Err(e) => eprintln!("{e}"),
+        }
+        // Need to wait until testbed server starts
+        std::thread::sleep(std::time::Duration::new(1, 0));
```

Comment:
I would prefer to have this step in `new_env` when we start the `testbed` server.

Reply:
But how can I detect that the testbed isn't start ? :thinking:

Reply:
A classic trick is to wait for the server to print to stdout, that's what we did in `run_testenv.py`:
https://github.com/Scille/parsec-cloud/blob/b4631f0562b6a9f2f5ce9a51e5e49f8df4421dd1/tests/scripts/run_testenv.py#L194-L198

Can't you do the same thing here?

Reply:
Done